pytorch_lightning/trainer/trainer.py

```diff
@@ -334,7 +337,14 @@ def __init__(
         self.checkpoint_connector = CheckpointConnector(self, resume_from_checkpoint)
         self.slurm_connector = SLURMConnector(self)
         self.tuner = Tuner(self)
-        self.train_loop = TrainLoop(self, max_epochs, min_epochs, max_steps, min_steps, num_sanity_val_steps)
+
+        if NEW_LOOP:
+            self.train_loop = EpochLoop(min_epochs, max_epochs, min_steps, max_steps)
```

Comment:
this is not going to be backwards compatible for users that access `trainer.train_loop`

Reply:
Not completely, but it is also the outer loop (over the epochs). We could do some property forwarding here, but on the other side I am thinking, it would be better not to do it to not mix those two concepts. Why would you even access the loop before if you couldn't even customize a thing?

Reply:
removed!